### PR TITLE
Download config from target on scan-pull-request

### PR DESCRIPTION
**Pull request by omerzi**
- [ ] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
---

### CODE REVIEW

File: commands/utils/params.go [hunk 1]
```diff
@@ -18,6 +20,8 @@ const (
 	FrogbotConfigFile = "frogbot-config.yml"
 )
 
+var errFrogbotConfigNotFound = fmt.Errorf("%s wasn't found in the Frogbot directory and its subdirectories. Continuing with environment variables", FrogbotConfigFile)
```
Comment (eyalbe4) [inline commands/utils/params.go:23]: ...Continuing with environment variables --> ...Assuming all the configuration is stored as envuronment variables.

File: commands/utils/params.go [hunk 2]
```diff
@@ -85,9 +89,27 @@ func GetParamsAndClient() (configAggregator FrogbotConfigAggregator, server *cor
 		return nil, nil, nil, err
 	}
 
-	configData, err := ReadConfig(frogbotConfigPath)
+	var targetConfigContent []byte
+	var configData *FrogbotConfigAggregator
+	var missingConfigErr bool
+	if gitParams.GitProvider.String() == vcsutils.GitHub.String() || gitParams.GitProvider.String() == vcsutils.GitLab.String() {
```
Comment (yahavi) [inline commands/utils/params.go:95]: I think we can remove the "String()" 
```suggestion
	if gitParams.GitProvider == vcsutils.GitHub || gitParams.GitProvider == vcsutils.GitLab {
```

File: commands/utils/params.go [hunk 3]
```diff
@@ -85,9 +89,27 @@ func GetParamsAndClient() (configAggregator FrogbotConfigAggregator, server *cor
...
+		if targetConfigContent != nil {
+			if err = yaml.Unmarshal(targetConfigContent, &configData); err != nil {
+				return nil, nil, nil, err
+			}
+		}
+	}
+	if targetConfigContent == nil {
+		configData, err = ReadConfig(frogbotConfigPath)
+		_, missingConfigErr = err.(*ErrMissingConfig)
+	}
```
Comment (yahavi) [inline commands/utils/params.go:110]: Let's export this logic to a new function "getFrogbotConfig"

Comment (eyalbe4) [inline commands/utils/params.go:110]: I suggest moving the above line into a "readConfig" function.

File: commands/utils/params.go [hunk 4]
```diff
@@ -347,3 +369,27 @@ func generateConfigAggregatorFromEnv(gitParams *Git, server *coreconfig.ServerDe
 	repo.Projects = append(repo.Projects, project)
 	return &FrogbotConfigAggregator{repo}, nil
 }
+
+func DownloadConfigFromTarget(client vcsclient.VcsClient) ([]byte, error) {
```
Comment (eyalbe4) [inline commands/utils/params.go:373]: Looks like this function can be made private.
Let's also add a description for this function.

**Review by yahavi — CHANGES_REQUESTED** [review]

File: commands/utils/params.go [hunk 5]
```diff
@@ -85,9 +89,27 @@ func GetParamsAndClient() (configAggregator FrogbotConfigAggregator, server *cor
...
+		if err != nil && !missingConfigErr {
+			return nil, nil, nil, err
+		}
+		if targetConfigContent != nil {
+			if err = yaml.Unmarshal(targetConfigContent, &configData); err != nil {
+				return nil, nil, nil, err
+			}
+		}
+	}
+	if targetConfigContent == nil {
```
Comment (eyalbe4) [inline commands/utils/params.go:107]: Let's explain this if statement.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: commands/utils/params.go (hunk 3 above)
Comment (omerzi) [reply commands/utils/params.go:110]: done

File: commands/utils/params.go (hunk 2 above)
Comment (omerzi) [reply commands/utils/params.go:95]: done

File: commands/utils/params.go (hunk 3 above)
Comment (omerzi) [reply commands/utils/params.go:110]: done

File: commands/utils/params.go (hunk 4 above)
Comment (omerzi) [reply commands/utils/params.go:373]: done

File: commands/utils/params.go (hunk 5 above)
Comment (omerzi) [reply commands/utils/params.go:107]: done

**Pull request merged**